core-patch/src/main/java/eu/neverblink/jelly/core/patch/internal/PatchDecoderImpl.java

```diff
@@ -0,0 +1,355 @@
+            return JellyOptions.SMALL_PREFIX_TABLE_SIZE;
+        }
+
+        return currentOptions.getMaxPrefixTableSize();
+    }
+
+    @Override
+    protected int getDatatypeTableSize() {
+        if (currentOptions == null) {
+            return JellyOptions.SMALL_DT_TABLE_SIZE;
+        }
+
+        return currentOptions.getMaxDatatypeTableSize();
+    }
+
+    @Override
+    public RdfPatchOptions getPatchOptions() {
+        return currentOptions;
+    }
+
+    @Override
+    public void ingestRow(RdfPatchRow row) {
+        switch (row.getRowCase()) {
+            case OPTIONS -> handleOptions(row.getOptions());
+            case STATEMENT_ADD -> handleStatementAdd(row.getStatementAdd());
+            case STATEMENT_DELETE -> handleStatementDelete(row.getStatementDelete());
+            case NAMESPACE_ADD -> handleNamespaceAdd(row.getNamespaceAdd());
+            case NAMESPACE_DELETE -> handleNamespaceDelete(row.getNamespaceDelete());
+            case TRANSACTION_START -> handleTransactionStart();
+            case TRANSACTION_COMMIT -> handleTransactionCommit();
+            case TRANSACTION_ABORT -> handleTransactionAbort();
+            case NAME -> handleName(row.getName());
+            case PREFIX -> handlePrefix(row.getPrefix());
+            case DATATYPE -> handleDatatype(row.getDatatype());
+            case HEADER -> handleHeader(row.getHeader());
+            case PUNCTUATION -> handlePunctuation();
+            case ROW_NOT_SET -> throw new RdfProtoDeserializationError(
+                "Row kind is not set or unknown: " + row.getRowCase()
+            );
+        }
+    }
+
+    @Override
+    public void ingestFrame(RdfPatchFrame frame) {
+        for (final var row : frame.getRowsList()) {
+            ingestRow(row);
+        }
+
+        if (isFrameStreamType) {
+            patchHandler.punctuation();
+        }
+    }
+
+    protected abstract void handleStatementAdd(RdfQuad statement);
+
+    protected abstract void handleStatementDelete(RdfQuad statement);
+
+    protected void handleOptions(RdfPatchOptions opt) {
+        JellyPatchOptions.checkCompatibility(opt, supportedOptions);
+        setPatchOptions(opt);
+    }
+
+    private void handleNamespaceAdd(RdfPatchNamespace nsRow) {
+        var valueIri = RdfTerm.from(nsRow.getValue());
+        var graphIri = RdfTerm.from(nsRow.hasGraph() ? nsRow.getGraph() : null);
+        patchHandler.addNamespace(
+            nsRow.getName(),
+            nameDecoder.provide().decode(valueIri.prefixId(), valueIri.nameId()),
+            decodeNsIri(graphIri)
+        );
+    }
+
+    private void handleNamespaceDelete(RdfPatchNamespace nsRow) {
+        var valueIri = RdfTerm.from(nsRow.hasValue() ? nsRow.getValue() : null);
+        var graphIri = RdfTerm.from(nsRow.hasGraph() ? nsRow.getGraph() : null);
+        patchHandler.deleteNamespace(nsRow.getName(), decodeNsIri(valueIri), decodeNsIri(graphIri));
+    }
+
+    private void handleTransactionStart() {
```

Comment:
These trivial private methods are useless. Please just call patchHandler.XXX() instead of them.

Reply:
Doesnt hurt to have it nice and consistent :)

Reply:
eh, okay then